src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -130,41 +115,32 @@ public virtual void Dispose()
         /// <inheritdoc />
         public ConsensusRuleEngine Register()
         {
-            Guard.Assert(this.Network.Consensus.Rules.Any());
-
-            foreach (ConsensusRule consensusRule in this.Network.Consensus.Rules)
-            {
-                consensusRule.Parent = this;
-                consensusRule.Logger = this.loggerFactory.CreateLogger(consensusRule.GetType().FullName);
-                consensusRule.Initialize();
-
-                this.consensusRules.Add(consensusRule.GetType().FullName, consensusRule);
+            this.headerValidationRules = this.Network.Consensus.HeaderValidationRules.Select(x => x as SyncConsensusRule).ToList();
```

Comment:
I don't understand this

Reply:
Answered on slack. Copypasted here: 
we need typecasts because interface is empty
But we need to call run or run async
So we are doing this ugly thing